pulser-core/pulser/sequence/helpers/_switch_device.py

```diff
@@ -117,20 +119,41 @@ def check_channels_match(
             if new_ch_obj.eom_config is None:
                 return (" with an EOM configuration.", "")
             if strict:
-                if (
-                    not seq.is_parametrized()
-                    and new_ch_obj.eom_config.mod_bandwidth
-                    != cast(RydbergEOM, old_ch_obj.eom_config).mod_bandwidth
-                ):
-                    return (
-                        "",
-                        " with the same mod_bandwidth for the EOM.",
-                    )
-                if (
-                    seq.is_parametrized()
-                    and new_ch_obj.eom_config != old_ch_obj.eom_config
-                ):
-                    return ("", " with the same EOM configuration.")
+                if not seq.is_parametrized():
+                    if (
+                        new_ch_obj.eom_config.mod_bandwidth
+                        != cast(
+                            RydbergEOM, old_ch_obj.eom_config
+                        ).mod_bandwidth
+                    ):
+                        return (
+                            "",
+                            " with the same mod_bandwidth for the EOM.",
+                        )
+                else:
+                    # Eom configs have to match is Sequence is parametrized
+                    new_eom_config = asdict(new_ch_obj.eom_config)
+                    old_eom_config = asdict(old_ch_obj.eom_config)
+                    # However, multiple_beam_control only matters when
+                    # the two beams are controlled
+                    if len(old_ch_obj.eom_config.controlled_beams) == 1:
+                        new_eom_config.pop("multiple_beam_control")
+                        old_eom_config.pop("multiple_beam_control")
+                        # Controlled beams only matter when only one beam
+                        # is controlled by the new eom
+                        if len(new_ch_obj.eom_config.controlled_beams) > 1:
+                            new_eom_config.pop("controlled_beams")
+                            old_eom_config.pop("controlled_beams")
```

Comment:
This is unreachable, I don't think this is what you want to do

Reply:
Sorry, I missed that up top it is `old_ch_obj` and then it's `new_ch_obj`, this is fine